engine/classes/Elgg/Application.php

```diff
@@ -0,0 +1,325 @@
+			'metastrings.php',
+			'navigation.php',
+			'notification.php',
+			'objects.php',
+			'output.php',
+			'pagehandler.php',
+			'pageowner.php',
+			'pam.php',
+			'plugins.php',
+			'private_settings.php',
+			'relationships.php',
+			'river.php',
+			'sessions.php',
+			'sites.php',
+			'statistics.php',
+			'system_log.php',
+			'tags.php',
+			'user_settings.php',
+			'users.php',
+			'upgrade.php',
+			'views.php',
+			'widgets.php',
+
+			// backward compatibility
+			'deprecated-1.7.php',
+			'deprecated-1.8.php',
+			'deprecated-1.9.php',
+			'deprecated-1.10.php',
+		);
+
+		// isolate global scope
+		call_user_func(function () use ($lib_dir, $lib_files) {
+
+			$setups = array();
+
+			// include library files, capturing setup functions
+			foreach ($lib_files as $file) {
+				$setup = (require_once "$lib_dir/$file");
+
+				if ($setup instanceof \Closure) {
+					$setups[$file] = $setup;
+				}
+			}
+
+			$events = $this->services->events;
+			$hooks = $this->services->hooks;
+
+			// run setups
+			foreach ($setups as $func) {
+				$func($events, $hooks);
+			}
+		});
+	}
+
+	/**
+	 * Bootstraps the Elgg engine.
+	 *
+	 * This method loads the full Elgg engine, checks the installation
+	 * state, and triggers a series of events to finish booting Elgg:
+	 * 	- {@elgg_event boot system}
+	 * 	- {@elgg_event init system}
+	 * 	- {@elgg_event ready system}
+	 *
+	 * If Elgg is fully uninstalled, the browser will be redirected to an
+	 * installation page.
+	 *
+	 * @see install.php
+	 * @return void
+	 */
+	public function bootCore() {
+		$config = $this->services->config;
+
+		if ($config->getVolatile('boot_complete')) {
+			return;
+		}
+
+		$this->loadSettings();
+
+		$config->set('boot_complete', false);
+
+		// This will be overridden by the DB value but may be needed before the upgrade script can be run.
+		$config->set('default_limit', 10);
+
+		$this->loadCore();
+
+		$events = $this->services->events;
+
+		// Connect to database, load language files, load configuration, init session
+		// Plugins can't use this event because they haven't been loaded yet.
+		$events->trigger('boot', 'system');
+
+		// Load the plugins that are active
+		$this->services->plugins->load();
+
+		// @todo move loading plugins into a single boot function that replaces 'boot', 'system' event
+		// and then move this code in there.
+		// This validates the view type - first opportunity to do it is after plugins load.
+		$viewtype = elgg_get_viewtype();
+		if (!elgg_is_registered_viewtype($viewtype)) {
+			elgg_set_viewtype('default');
+		}
+
+		// @todo deprecate as plugins can use 'init', 'system' event
+		$events->trigger('plugins_boot', 'system');
+
+		// Complete the boot process for both engine and plugins
+		$events->trigger('init', 'system');
+
+		$config->set('boot_complete', true);
+
+		// System loaded and ready
+		$events->trigger('ready', 'system');
+	}
+
+	/**
+	 * Get the database instance (also loads settings if not yet loaded)
```

Comment:
I added this because devs are currently crudely reading settings directly and using ext/mysql (fails to support multi-DB config files). This makes it easy to get a DB connection w/o loading core yet still inject the full logger during full boot.